src/org/freenetproject/freemail/ui/web/SettingsToadlet.java

```diff
@@ -0,0 +1,112 @@
+package org.freenetproject.freemail.ui.web;
+
+import freenet.clients.http.PageNode;
+import freenet.clients.http.ToadletContainer;
+import freenet.clients.http.ToadletContext;
+import freenet.pluginmanager.PluginRespirator;
+import freenet.support.HTMLNode;
+import freenet.support.api.HTTPRequest;
+import org.freenetproject.freemail.config.Configurator;
+import org.freenetproject.freemail.l10n.FreemailL10n;
+
+import javax.naming.SizeLimitExceededException;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.HashMap;
+
+public class SettingsToadlet extends WebPage {
+
+	private static final String PATH = WebInterface.PATH + "/Settings";
+
+	private final Configurator config;
+
+	private final ToadletContainer toadletContainer;
+
+	SettingsToadlet(PluginRespirator pluginRespirator, LoginManager loginManager, Configurator config,
+									ToadletContainer toadletContainer) {
+		super(pluginRespirator, loginManager);
+		this.config = config;
+		this.toadletContainer = toadletContainer;
+	}
+
+	@Override
+	HTTPResponse makeWebPageGet(URI uri, HTTPRequest req, ToadletContext ctx, PageNode page) throws IOException {
+		HTMLNode settingsBox = addInfobox(page.content, FreemailL10n.getString("Freemail.SettingsToadlet.title"));
+		addChild(settingsBox, "settings-form",
+			 new HashMap<String, String>() {{
```

Comment:
Please move the creation of the `Map` into its own method, the anonymous subclass should not be necessary then anymore.